src/components/FileUpload.js

```diff
@@ -140,66 +178,73 @@ class FileUpload extends React.Component {
   };
 
   _validateImageDimensions = (width, height) => {
-    const validationMessages = [];
+    const failedImageValidationTypes = [];
 
-    if (this.props.onFileValidation && this.props.imageValidation) {
+    if (this.props.imageValidation) {
       if (this.props.imageValidation.exactHeight && this.props.imageValidation.exactHeight !== height) {
-        validationMessages.push('exact_height');
+        failedImageValidationTypes.push('exact_height');
       }
 
       if (this.props.imageValidation.exactWidth && this.props.imageValidation.exactWidth !== width) {
-        validationMessages.push('exact_width');
+        failedImageValidationTypes.push('exact_width');
       }
 
       if (this.props.imageValidation.maxHeight && this.props.imageValidation.maxHeight < height) {
-        validationMessages.push('max_height');
+        failedImageValidationTypes.push('max_height');
       }
 
       if (this.props.imageValidation.maxWidth && this.props.imageValidation.maxWidth < width) {
-        validationMessages.push('max_width');
+        failedImageValidationTypes.push('max_width');
       }
 
       if (this.props.imageValidation.minHeight && this.props.imageValidation.minHeight > height) {
-        validationMessages.push('min_height');
+        failedImageValidationTypes.push('min_height');
       }
 
       if (this.props.imageValidation.minWidth && this.props.imageValidation.minWidth > width) {
-        validationMessages.push('min_width');
+        failedImageValidationTypes.push('min_width');
       }
 
       if (this.props.imageValidation.ratioHeight && this.props.imageValidation.ratioWidth && (this.props.imageValidation.ratioHeight / this.props.imageValidation.ratioWidth !== height / width)) {
-        validationMessages.push('image_ratio');
+        failedImageValidationTypes.push('image_ratio');
       }
     }
 
-    return validationMessages;
+    return failedImageValidationTypes;
   };
 
   _validateFile = (file, width = null, height = null) => {
-    const validationMessages = this._validateImageDimensions(width, height);
+    let failedValidationTypes = [];
+    const fileExt = file.name.split('.').pop();
     const isTooBig = this.props.maxFileSize < file.size / 1000;
-    const isInvalidType = this.props.allowedFileTypes && this.props.allowedFileTypes.indexOf(file.type) < 0;
+    const isInvalidFileType = this.props.allowedFileTypes && this.props.allowedFileTypes.indexOf(file.type) < 0;
+    const isInvalidFileExtension = this.props.allowedFileTypes && this.props.allowedFileTypes.indexOf(fileExt) < 0;
 
-    if (this.props.onFileValidation) {
-      if (isTooBig) {
-        validationMessages.push('file_size');
-      }
+    if (isTooBig) {
+      failedValidationTypes.push('file_size');
+    }
 
-      if (isInvalidType) {
-        validationMessages.push('file_type');
-      }
+    if (isInvalidFileType && isInvalidFileExtension) {
```

Comment:
So I had a reason why `&&`..but I can't remember it and further thinking makes me believe I should just do `||`. Going to change